src/python/turicreate/test/test_api_visibility.py

```diff
@@ -0,0 +1,363 @@
+        assert set(actual) == set(expected + ['ItemContentRecommender'])
+
+        actual = get_visible_items(turicreate.recommender.popularity_recommender)
+        assert set(actual) == set(expected + ['PopularityRecommender'])
+
+        actual = get_visible_items(turicreate.recommender.util)
+        expected = [
+            'random_split_by_user',
+            'precision_recall_by_user',
+            'compare_models'
+        ]
+
+        check_visible_modules(actual, expected)
+
+    def test_nearest_neighbors(self):
+        actual = get_visible_items(turicreate.nearest_neighbors)
+        expected = ['NearestNeighborsModel', 'create']
+        check_visible_modules(actual, expected)
+
+    def test_clustering(self):
+        actual = get_visible_items(turicreate.clustering)
+        expected = ['kmeans', 'dbscan']
+        check_visible_modules(actual, expected)
+
+    def test_topic_model(self):
+        actual = get_visible_items(turicreate.topic_model)
+        expected = ['topic_model',
+                    'create',
+                    'perplexity']
+        check_visible_modules(actual, expected)
+
+    def test_text_analytics(self):
+        actual = get_visible_items(turicreate.text_analytics)
+        expected = ['tf_idf',
+                    'bm25',
+                    'stop_words',
+                    'count_words',
+                    'count_ngrams',
+                    'random_split',
+                    'parse_sparse',
+                    'parse_docword',
+                    'tokenize',
+                    'drop_words']
+
+        check_visible_modules(actual, expected)
+
+    def test_classifier(self):
+        actual = get_visible_items(turicreate.classifier)
+        expected = ['create',
+                    'logistic_classifier',
+                    'nearest_neighbor_classifier',
+                    'random_forest_classifier',
+                    'decision_tree_classifier',
+                    'boosted_trees_classifier',
+                    'decision_tree',
+                    'svm_classifier',
+                    'nearest_neighbor_classifier']
+
+        check_visible_modules(actual, expected)
+
+    def test_regression(self):
+        actual = get_visible_items(turicreate.regression)
+        expected = ['create',
+                    'linear_regression',
+                    'boosted_trees_regression',
+                    'decision_tree_regression',
+                    'random_forest_regression']
+
+        check_visible_modules(actual, expected)
+
+    def test_toolkits(self):
+        actual = get_visible_items(turicreate.toolkits)
+
+        ## TODO: LazyModule has hidden some of these imports for us. We need to
+        ##       fix this somehow. Works perfectly in the Repel.
+        
+        expected = [
+                    "classifier",
+                    "clustering",
+                    "distances",
+                    "evaluation",
+                    "graph_analytics",
```

Comment:
Shouldn't there be more modules visible. Are you sure this is actually testing the right thing.

Reply:
Yeah, they should be. The `TODO` was more of a mental note for myself that I needed to add more toolkits to this test. I can add them now.